lsp/server/src/utils/workspaceUtils.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { SFDX_PROJECT_FILE } from './constants';
+import * as fs from 'fs';
+import * as path from 'path';
+import { WorkspaceFolder } from 'vscode-languageserver';
+
+export class WorkspaceUtils {
+    private static workspaceFolders: WorkspaceFolder[] | undefined | null = [];
+
+    static readonly DEFAULT_APP_PATH = path.join(
+        'force-app',
+        'main',
+        'default'
+    );
+
+    static readonly LWC_PATH = path.join(
+        WorkspaceUtils.DEFAULT_APP_PATH,
+        'lwc'
+    );
+
+    public static setWorkSpaceFolders(
+        workSpaceFolders: WorkspaceFolder[] | null | undefined
+    ) {
+        this.workspaceFolders = workSpaceFolders;
+    }
+
+    // LSP server has a different way to fetch WorkSpaceDir than root project which relis on vscode workspace.
+    static getWorkspaceDir(): string {
```

Comment:
This method gets created only inside this file. It throws error when there is no workspace. But the caller of this method catches that error and returns false. Which means, this error is serving no purpose. 

There are number of ways to clean up this code. I'll list one here.

You can get rid of `getWorkspaceDir()`. And assuming that you're mashing `lwcFolderExists()` and `isSfdxProjectOpened()` into one function, then, before you call `fs.existsSync()` in there you can do the same conditional
```
if (!this.workspaceFolders || this.workspaceFolders.length === 0) {
```
and see if you can do early exit if no valid workspace folders have been set yet.